plugin/build_baron.go

```diff
@@ -47,45 +50,78 @@ func (bbp *BuildBaronPlugin) Configure(conf map[string]interface{}) error {
 		}
 		webhookConfigured := webHook.Endpoint != ""
 		if !webhookConfigured && proj.TicketCreateProject == "" {
-			return fmt.Errorf("ticket_create_project and taskAnnotationSettings.FileTicketWebHook endpoint cannot both be blank")
```

Comment:
now that we no longer have the guarantee that they won't both be defined, should we change how the IsWebhookConfigured() function behaves? For example, if a project has both the webhook configured and the TicketCreateProject defined, then we don't know where to file the ticket if someone clicks on the button. So maybe we should add a check like this to IsWebhookConfigured() and error if both are defined.

Reply:
We will still have a guarantee that they wont both be defined, since we are not removing the build baron validation coming from the admin page until we remove the ability to define plugin configs on the admin page.  I think your comment is right and should be implemented in https://jira.mongodb.org/browse/EVG-15419, however I think the scope for now is to stop the app from crashing when a bad build baron config is processed.

Reply:
Where do we have the guarantee that they won't both be defined? 
While the main scope is to stop the app from crashing, I wouldn't want to potentially leave anything in a potentially bad state.

Reply:
We have kept line 107 that checks if both are defined and sends a critical error if so. None of the previously existing validation checks have changed, only how we handle the errors has changed.

Reply:
right, but the way we used to handle it is to not start the app, so if someone had both defined, no user would be clicking a "file ticket" button because we wouldn't have an app running at all. Now that the way we handle it no longer provides that guarantee, I think we should add a check to IsWebhookConfigured() so that when a user tries to file a ticket in that state, they  gets an error saying that it's not configured properly, instead of it picking one and filing it with one, when it may not be the one they intended. 
A critical error will alert us, but nothing will get fixed until someone on our team sees it, understands whats wrong, reaches out to the team, and fixes it.